### PR TITLE
Zoom to Mouse Cursor and Only pan deltaY on vertical scale fix

### PR DESCRIPTION
**Pull request by lysaght**
Zoom to Mouse Cursor (Draft) as suggested in Issue #21

### CODE REVIEW

File: src/chart.zoom.js [hunk 1]
```diff
@@ -41,32 +41,36 @@ function zoomIndexScale(scale, zoom, center) {
 
 }
 
-
 function zoomTimeScale(scale, zoom, center) {
 	var options = scale.options;
-	var minDelta, maxDelta;
-	var newDiff;
+	var range = scale.right - scale.left;
```
Comment (etimberg) [inline src/chart.zoom.js]: This should account for the vertical case too

**Comment by etimberg** [comment]
Code looks good. Will test this out as well.

**Comment by lysaght** [comment]
Also added fix for Issue #23  (Vertical Pan pans a little horizontally as well).

File: src/chart.zoom.js (hunk 1 above)
Comment (lysaght) [inline src/chart.zoom.js]: Does it need to? http://www.chartjs.org/docs/#scales-time-scale says Time Scale can only be placed on the X axis.

Comment (etimberg) [inline src/chart.zoom.js]: I think that will change when v2.2.0 of Chart.js is released. I merged a PR that adds support for the category and time axes on the left/right

Comment (lysaght) [inline src/chart.zoom.js]: OK.  I've added a draft implementation for Time scale 'Zoom to Cursor' on the Vertical.  Wasn't able to test though as I couldn't work out how to configure a Chart.js example with Time scale on the Vertical.

**Comment by etimberg** [comment]
Thanks @lysaght 
great work!

**Pull request merged**